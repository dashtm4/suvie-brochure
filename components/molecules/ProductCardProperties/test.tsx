import { shallow } from 'enzyme'
import toJSON from 'enzyme-to-json'

import { ProductCardProperties } from '.'

describe('<ProductCardProperties />', () => {
  const title = 'Test title'
  const properties = [{ item: 'Property 1' }, { item: 'Property 2' }]

  it('should renders correct <ProductCardProperties />', () => {
    const wrapper = shallow(
      <ProductCardProperties title={title} properties={properties} />
    )

    expect(wrapper.contains(title)).toBeTruthy()
    expect(wrapper.contains(properties[0].item)).toBeTruthy()
    expect(wrapper.contains(properties[1].item)).toBeTruthy()

    expect(toJSON(wrapper)).toMatchSnapshot()
  })

  it('should has className from prop', () => {
    const className = 'test-class-name'

    const wrapper = shallow(
      <ProductCardProperties
        title={title}
        properties={properties}
        className={className}
      />
    )

    expect(wrapper.prop('className')).toContain(className)

    expect(toJSON(wrapper)).toMatchSnapshot()
  })
})
