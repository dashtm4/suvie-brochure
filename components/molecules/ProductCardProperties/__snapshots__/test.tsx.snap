// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductCardProperties /> should has className from prop 1`] = `
<div
  className="test-class-name"
>
  <h3
    className="font-secondary mb-1"
  >
    Test title
  </h3>
  <ul
    className="text-octonary"
  >
    <li
      key="Property 1"
    >
      Property 1
    </li>
    <li
      key="Property 2"
    >
      Property 2
    </li>
  </ul>
</div>
`;

exports[`<ProductCardProperties /> should renders correct <ProductCardProperties /> 1`] = `
<div>
  <h3
    className="font-secondary mb-1"
  >
    Test title
  </h3>
  <ul
    className="text-octonary"
  >
    <li
      key="Property 1"
    >
      Property 1
    </li>
    <li
      key="Property 2"
    >
      Property 2
    </li>
  </ul>
</div>
`;
